src/libraries/System.Private.CoreLib/src/System/Threading/Thread.OSX.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal static class ThreadOSX
+    {
+        private static bool CheckEnableAutoreleasePool()
+        {
+#if !CORECLR
```

Comment:
Why this ifdef? This switch was enabled for threadpool for all runtimes before this change.

Reply:
This is an implementation detail that needs to be decided. My starting premise here is `ThreadPool` threads shouldn't be special cased in any runtime. In this case the type originally used, `AppContextConfigHelper`, works fine but only after the implementing runtime has populated it. This works with `ThreadPool` because it wasn't needed so early. In order to read the needed setting earlier enough in CoreCLR, we need to use the `CLRConfig` type. I can fallback to `AppContextConfigHelper` for the non-CoreCLR scenario but that technically work as expected until Mono respects the feature for all threads. At present it would only work for `ThreadPool` threads for all runtimes and not all threads so I think this behavior is the correct one until that work happens in Mono.